progress/__init__.py

```diff
@@ -95,8 +96,8 @@ def __init__(self, *args, **kwargs):
 
     @property
     def eta(self):
-        avg = self.avg
-        return int(ceil(self.remaining / avg)) if avg else 0
+        if not self.index: return 0
+        return int(ceil((self.max - self.index) * self.elapsed/self.index))
```

Comment:
I actually use right now a ETA confidence interval implementation discussed in this stack overflow post: http://stackoverflow.com/questions/933242/smart-progress-bar-eta-computation.  I also found the method described by SPWorley in that post to be better than a simple average.